docs/RAG_serving_demo.md

```diff
@@ -0,0 +1,155 @@
+
+Here are the steps we are going to follow:
+- building the container image of the model server togather with required python dependencies (public image has minimal server of python components)
+- adding the RAG servable configuration to a configmap (it includes the model server configuraiton, MediaPipe graph, python code and also a list of documents to be analysed)
+- optionally adding Hugging Faces token to a secret, it will be attached to the deployed service to pull the models in case they requires authentication
+- deploying the service via the operator and ModelServer custom resource
+- running the client to query the document
+- online updates of the documents and reruning the client query
+
+
+## Building the container image of the model server
+
+The public image of Model Server is enabled for python execution but it has a minimal list of python dependencies. Depending on the use cases, a custom dependencies might need to be included in the image. Follow the instructions from https://github.com/openvinotoolkit/model_server/tree/main/demos/python_demos/rag_chatbot
+
+```
+git clone https://github.com/openvinotoolkit/model_server.git
+cd model_server
+# for ubi base image
+make python_image BASE_OS=redhat OVMS_CPP_DOCKER_IMAGE=registry.connect.redhat.com/intel/openvino-model-server OVMS_CPP_IMAGE_TAG=2024.1-gpu 
+# or for ubuntu22 base image
+make python_image BASE_OS=redhat OVMS_CPP_DOCKER_IMAGE=openvino/model_server OVMS_CPP_IMAGE_TAG=2024.1-gpu
+```
+Push the build image to your image registry.
+
+In case of OpenShift, this build operation can be automated using a BuildConfig resource.
+
+```
+spec:
+  nodeSelector: null
+  output:
+    to:
+      kind: ImageStreamTag
+      name: 'ovms:py'
+  resources: {}
+  strategy:
+    type: Docker
+    dockerStrategy:
+      dockerfilePath: Dockerfile.redhat
+      buildArgs:
+        - name: IMAGE_NAME
+          value: 'registry.connect.redhat.com/intel/openvino-model-server@sha256:b8721a65da98bd9b354680ecb2dca8a4d60a1097babe9822ab264f8e951190f7'
+  postCommit: {}
+  source:
+    type: Git
+    git:
+      uri: 'https://github.com/openvinotoolkit/model_server'
+      ref: main
+    contextDir: demos/python_demos
+```
+It will trigger the build based on provided above Dockerfile and the based image. That image will be automatically pushed to the local cluster container registry in the OpenShift.
+
+## Adding the RAG servable configuration to a configmap
+
+The Model Server configuration including servable configuration needs to be attached to deployed pods. In this demo they will be passed via a configmap which can stored arbitrary files. The operator automatically mounts the contnent of a configmap to the serving pod.
+
+```
+pushd $(pwd)
+cd model_server/demos/python_demos/rag_chatbot/servable_stream
+echo "https://gist.githubusercontent.com/ryanloney/42b8ebe29f95ebd4382ee0b2bb50bea2/raw/cfbb679fefb6babec675c7806254a5fff29a5e6b/aipc.txt" > docs.txt
```

Comment:
Will this file be persistent?

Reply:
no, it is expected to be changed

Reply:
maybe we should keep this file as an example in operator repository?